src/ios-scroll.js

```diff
@@ -0,0 +1,557 @@
+    // scrollTop to match where we syntheticly scrolled (else, the user
+    // wouldn't be able to natively scroll up because Safari thinks
+    // they're "at" the top of the doc).
+    this.scrollTop_ += this.offset_;
+    this.min_ -= offset;
+    this.max_ -= offset;
+    this.scrollTo_(0);
+    // Suppress this body scrolled event, we expect it.
+    this.bodyScrolled_ = null;
+    this.scrollMoveEl_.style.transform = 'translateY(' + this.scrollTop_ + 'px)';
+    this.scrollMoveEl_./*OK*/scrollIntoView(true);
+
+    this.setIframeInteractionAllowed_(true);
+  }
+
+  /**
+   * Enables or disables interaction with iframes.
+   *
+   * @param {boolean} allowed
+   */
+  setIframeInteractionAllowed_(allowed) {
+    const value = allowed ? '' : 'none';
+    const styles = this.iframeStyles_;
+    for (let i = 0; i < styles.length; i++) {
+      styles[i].pointerEvents = value;
+    }
+  }
+
+  /**
+   * A callback bound to the `touchstart` event on the document.
+   *
+   * @param {TouchEvent} event
+   */
+  touchstart_(event) {
+    // We don't allow zooming
+    if (event.touches.length > 1) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    if (this.autoScrolling_) {
+      cancelAnimationFrame(this.autoScrolling_);
+      this.autoScrolling_ = 0;
+    } else {
+      this.scrollStarted_();
+    }
+
+    this.velocity_ = 0;
+    this.amplitude_ = 0;
+    this.timestamp_ = Date.now();
+    this.reference_ = this.eventY_(event);
+  }
+
+  /**
+   * A callback bound to the `touchmove` event on the document.
+   *
+   * @param {TouchEvent} event
+   */
+  touchmove_(event) {
+    // We don't allow zooming
+    if (event.touches.length > 1) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    const y = this.eventY_(event);
+    const delta = this.reference_ - y;
+    this.track_(delta);
+    this.reference_ = y;
+    this.scrollTo_(this.offset_ + delta);
+
+    event.preventDefault();
+    event.stopPropagation();
+    return false;
+  }
+
+  /**
+   * A callback bound to the `touchend` event on the document.
+   *
+   * @param {TouchEvent} event
+   */
+  touchend_(event) {
+    // We don't allow zooming
+    if (event.touches.length > 1) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    // Only simulate momentum if we think the user meant to have it.  Ie. if
+    // you hang your finger for a few milliseconds, you probably meant to stop
+    // the scrolling entirely.
+    const elapsed = Date.now() - this.timestamp_;
+    if (elapsed < SCROLL_INTENTION_ELAPSED_CUTOFF) {
+      // Determine where we should stop the synthetic momentum.
+      // The multiplier is arbitrary.
+      const multiplier = 1.1;
+      this.amplitude_ = multiplier * this.velocity_;
+      this.target_ = Math.round(this.offset_ + this.amplitude_);
+      this.autoScrolling_ = requestAnimationFrame(this.boundAutoScroll);
```

Comment:
No polyfill needed, iOS back to (at least) 8 has it. I'll update to use `win`, though.